R/process_spec.R

```diff
@@ -0,0 +1,174 @@
+#' data("raman_hdpe")
+#' plot(raman_hdpe)
+#'
+#' # Process spectra with range restriction and baseline subtraction
+#' process_spec(raman_hdpe,
+#'                 active_processing = TRUE,
+#'                 range_decision = TRUE,
+#'                 min_range = 500,
+#'                 max_range = 3000,
+#'                 baseline_decision = TRUE,
+#'                 baseline_selection = "Polynomial",
+#'                 baseline_polynomial = 8,
+#'                 derivative_decision = FALSE) |>
+#'   lines(col = "darkred")
+#'
+#' # Process spectra with smoothing and derivative
+#' process_spec(raman_hdpe,
+#'                 active_processing = TRUE,
+#'                 smooth_decision = TRUE,
+#'                 smooth_polynomial = 3,
+#'                 smooth_window = 11,
+#'                 derivative_decision = TRUE,
+#'                 derivative_order = 1,
+#'                 derivative_polynomial = 3,
+#'                 derivative_window = 11) |>
+#'   lines(col = "darkgreen")
+#'
+#' # Sampling a spectrum
+#' tiny_map <- read_any(read_extdata("CA_tiny_map.zip"))
+#' sampled <- sample_spec(tiny_map, size = 3)
+#' print(sampled)
+#' plot(sampled)
+#'
+#' @export
+process_spec <- function(x, ...) {
+  UseMethod("process_spec")
+}
+
+#' @rdname process_spec
+#'
+#' @export
+process_spec.default <- function(x, ...) {
+  stop("object 'x' needs to be of class 'OpenSpecy'")
+}
+
+#' @rdname process_spec
+#'
+#' @export
+process_spec.OpenSpecy <- function(x,
+                                   active_processing = T,
+                                   adj_intensity_decision = F,
+                                   type = "none",
+                                   conform_decision = T,
+                                   new_wavenumbers = NULL,
+                                   res = 5,
+                                   range_decision = F,
+                                   min_range = 0,
+                                   max_range = 6000,
+                                   flatten_decision = F,
+                                   flatten_min = 2200,
+                                   flatten_max = 2420,
+                                   smooth_decision = F,
+                                   smooth_polynomial = 3,
+                                   smooth_window = 11,
+                                   baseline_decision = F,
+                                   baseline_selection = "Polynomial",
+                                   raw_baseline = F,
+                                   baseline_polynomial = 8,
+                                   wavenumber_fit = NULL,
+                                   intensity_fit = NULL,
+                                   derivative_decision = T,
+                                   derivative_order = 1,
+                                   derivative_polynomial = 3,
+                                   abs = T,
+                                   derivative_window = 11, ...) {
```

Comment:
This is quite a number of parameters. We could rearrange them in lists like that to make it clearer:

```suggestion
                                   active_processing = T,
                                   adj_intensity = F,
                                   adj_intensity_pars = list(type = "none"),
                                   conform = T,
                                   conform_pars = list(new_wavenumbers = NULL, res = 5,
                                   ...
```

What do you think?

Reply:
I agree that it makes it clearer, and more tidy, all for it! Great idea.